### PR TITLE
Nitro server support

### LINKED ISSUE #14: Expose $clientFetch within Nuxt server (Nitro) API
This is perhaps too much to ask/out of scope of the project, but nevertheless:

Currently, the helpers created by nuxt-open-fetch are not available in Nuxt server handlers (in particular, they're not referenced by `.nuxt/tsconfig.server.json`). That would be great if `$clientFetch` were there, same as `$fetch` is.

### PR DESCRIPTION
**Pull request by skf-funzt**
# Features

- [x] **Nitro Server support** :point_right:  resolves #14 
- [x] **Automatic API route creation from OpenAPI schemas**
- [x] Updated tests

## Usage - Nitro Server support

`nuxt.config.ts`
```ts
export default defineNuxtConfig({
  ...
  openFetch: {
    clients: {
      pets: {
        baseURL: 'https://petstore3.swagger.io/api/v3'
      },
    },
    // Adding server support
    // ! The baseURL  needs to be set properly via config or the schema
    servers: {
      pets: {
        baseURL: 'https://petstore3.swagger.io/api/v3'
      },
    },
  },
  ...
})
```

`server/api/pets.ts`
```ts
export default defineEventHandler(async (event) => {
  const { $fetchPets } = useNuxtOpenFetchServer()
  return await $fetchPets ("/pet/{petId}", {
    path: {
      petId: 1,
    },
  });
})
```

Using automatic server API routes
```ts
const {
  data: usePetsFetchServerData,
  execute: usePetsFetchServerExecute,
  error: usePetsFetchServerError,
} = await usePetsFetchServer("/pet/{petId}", {
  immediate: false,
  path: {
    petId: 1,
  },
});
```

## Improvements

- [x] Adding server routes from the schema automatically
- [x] A custom fetch interface to resolve local server routes the same way as the use[$client}Fetch approach

## Missing

- [x] Add proper type support for server plugin - especially for the created functions
- [ ] Add documentation for server support
- [ ] Add usage example
- [ ] Working CI tests - currently only `test:watch` is working properly

### CODE REVIEW

File: playground/server/api/pets.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  const data = await $fetchPets ("/pet/{petId}", {
+    path: {
+      petId: 1,
+    },
+  });
+
+  return ({
+    data,
+  })
+})
```
Comment (IlyaSemenov) [inline playground/server/api/pets.ts:12]: Please lint this (get rid of 'no new line at the end of file'). The project uses [editorconfig](https://github.com/enkot/nuxt-open-fetch/blob/18c4ef73a106f95378fd4313ebeb1ef0c25de22a/.editorconfig#L9) but its setup was ignored.

File: src/runtime/nuxt-plugin.ts [hunk 2]
```diff
@@ -12,3 +12,4 @@ export default defineNuxtPlugin(() => {
     }), {})
   }
 })
+
```
Comment (IlyaSemenov) [inline src/runtime/nuxt-plugin.ts:15]: why the stray change?

Comment (skf-funzt) [reply src/runtime/nuxt-plugin.ts:15]: Don't know what you mean here.

**Comment by skf-funzt** [comment]
@enkot I added the feature for automatic server API route creation. Could you please have a look?

File: playground/server/api/pets.ts (hunk 1 above)
Comment (skf-funzt) [reply playground/server/api/pets.ts]: Thanks for the hint :+1:

**Comment by enkot** [comment]
Hi @skf-funzt 
Sorry for not replying for a long time. Thanks for your PR! I really appreciate it. 💚 
While the general idea is correct, my vision for the implementation is a bit different.
The idea is to have the same preconfigured client implementation in the Nuxt and Nitro context, same way we have plain  `$fetch`:
```ts
// Nuxt
const { $petsFetch } = useNuxtApp()

$petsFetch('/pet/{petId}', {
  path: {
    petId: 2
  }
})

// Nitro
const { $petsFetch } = useNitroApp()

$petsFetch('/pet/{petId}', {
  path: {
    petId: 2
  }
})
```
Because it requires quite a lot of changes changes in this branch, I created another PR - https://github.com/enkot/nuxt-open-fetch/pull/23. 
I will add you as a contributor there.

**Comment by skf-funzt** [comment]
@enkot 
What do you think about the automatic nitro route creation to avoid having CORS issues on the API's backend side?

**Comment by enkot** [comment]
I think handling CORS is beyond the scope of this module. NuxtOpenFetch simply extends `$fetch`/`useFetch` functionality with OpenAPI types.

**Comment by skf-funzt** [comment]
OK understood. Thanks for pointing that out.
I thought it would be nice to have an immediate solution for the CORS
issues with certain backends as I ran into the same issue and this was the
solution for me without using a different package.

**Comment by enkot** [comment]
@skf-funzt I noticed you made some regarding `/utils`. Just want to note that `utils` was removed, and functions for module and runtime are used in place now.  So it should work correctly now :)

**Pull request closed without merging**